cmd/clusterctl/cmd/config_provider.go

```diff
@@ -79,6 +79,7 @@ var configProviderCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) error {
 		return runGetComponents()
 	},
+	Deprecated: "use `clusterctl generate provider` instead",
```

Comment:
what is the plan for removal of the deprecated commands:
- when would that happen?
- is there an issue to track it?

Reply:
The plan is to support the old commands for `0.4` and to remove it in the subsequent release. I've created an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/4590) for tracking it. Do you think we should add a warning about removing it in the `Deprecated` description string?